crates/dbsp/src/operator/neighborhood.rs

```diff
@@ -484,37 +489,35 @@ mod test {
     fn test_circuit(
         circuit: &mut RootCircuit,
     ) -> AnyResult<(
-        InputHandle<bool>,
         InputHandle<Option<NeighborhoodDescr<i32, i32>>>,
         CollectionHandle<i32, (i32, i32)>,
         OutputHandle<OrdIndexedZSet<isize, (i32, i32), i32>>,
     )> {
-        let (reset_stream, reset_handle) = circuit.add_input_stream::<bool>();
         let (descr_stream, descr_handle) =
             circuit.add_input_stream::<Option<NeighborhoodDescr<i32, i32>>>();
         let (input_stream, input_handle) = circuit.add_input_indexed_zset::<i32, i32, i32>();
 
         let range_handle = input_stream
-            .neighborhood(&descr_stream, &reset_stream)
+            .neighborhood(&descr_stream)
+            .integrate()
+            .index()
             .output();
 
-        Ok((reset_handle, descr_handle, input_handle, range_handle))
+        Ok((descr_handle, input_handle, range_handle))
     }
 
     #[test]
-    fn bounded_range_test() {
-        let (mut dbsp, (reset_handle, descr_handle, input_handle, output_handle)) =
+    fn neighborhood_test() {
```

Comment:
are there tests when the type specified is wrong?
I guess you can do that using the handle API

Reply:
It wouldn't compile with wrong types.